device/samples/pnp/pnpTemperatureController.js

```diff
@@ -0,0 +1,256 @@
+const commandComponentCommandNameSeparator = '*';
+let intervalToken1;
+let intervalToken2;
+let intervalToken3;
+
+const commandNameGetMaxMinReport1 = thermostat1ComponentName + commandComponentCommandNameSeparator + 'getMaxMinReport';
+const commandNameGetMaxMinReport2 = thermostat2ComponentName + commandComponentCommandNameSeparator + 'getMaxMinReport';
+const commandNameReboot = 'reboot';
+
+const commandHandler = async (request, response) => {
+  helperLogCommandRequest(request);
+  switch (request.methodName) {
+  case commandNameGetMaxMinReport1: {
+    await sendCommandResponse(request, response, 200, 'max min report from thermostat 1');
+    break;
+  }
+  case commandNameGetMaxMinReport2: {
+    await sendCommandResponse(request, response, 200, 'max min report from thermostat 2');
+    break;
+  }
+  case commandNameReboot: {
+    await sendCommandResponse(request, response, 200, 'reboot response');
+    break;
+  }
+  default:
+    await sendCommandResponse(request, response, 404, 'unknown method');
+    break;
+  }
+};
+
+const sendCommandResponse = async (request, response, status, payload) => {
+  try {
+    await response.send(status, payload);
+    console.log('Response to method \'' + request.methodName +
+              '\' sent successfully.' );
+  } catch (err) {
+    console.error('An error ocurred when sending a method response:\n' +
+              err.toString());
+  }
+};
+
+const helperLogCommandRequest = (request) => {
+  console.log('Received command request for comand name \'' + request.methodName + '\'');
+
+  if (!!(request.payload)) {
+    console.log('The command request paylaod :');
+    console.log(request.payload);
+  }
+};
+
+
+const helperCreateReportedPropertiesPatch = (propertiesToReport, componentName) => {
+    let patch;
+    if (!!(componentName)) {
+      patch = { };
+      propertiesToReport.__t = "c";
+      patch[componentName] = propertiesToReport;
+    }
+    else {
+      patch = { };
+      patch = propertiesToReport;
+      }
+    if (!!(componentName)) {
+    console.log('The following properties will be updated for component:' + componentName);
+    }
+    else{
+      console.log('The following properties will be updated for root interface:');
+    }
+    console.log(patch);
+    return patch;
+  };
+
+const updateComponentReportedProperties = (deviceTwin, patch, componentName) => {
+  let logLine;
+  if (!!(componentName))
+  {
+    logLine = 'Properties have been reported for component:' + componentName;
+  }
+  else{
+    logLine = 'Properties have been reported for root interface';
+  }
+  deviceTwin.properties.reported.update(patch, function (err) {
+    if (err) throw err;
+    console.log(logLine);
+  });
+};
+
+const helperAttachHandlerForDesiredPropertyPatches = (deviceTwin) => {
+
+  deviceTwin.on('properties.desired', (delta) => {
+    if (delta[thermostat1ComponentName] || delta[thermostat2ComponentName] || delta[deviceInfoComponentName]) {
```

Comment:
> { [](start = 109, length = 2)

Inconsistent use of brace after the if.  (Although I prefer this it's not consistent with your other uses.)

Reply:
not sure what u mean here